src/lib/chips/chip-list-key-manager.ts

```diff
@@ -0,0 +1,89 @@
+import {QueryList} from '@angular/core';
+import {ListKeyManager} from '../core/a11y/list-key-manager';
+import {MdBasicChip} from './chip';
+
+/**
+ * Manages keyboard events for the chip list and its chips. When instantiated
+ * with a QueryList of MdBasicChip (i.e. any chip), it will ensure focus and
+ * keyboard navigation are properly handled.
+ */
+export class ChipListKeyManager extends ListKeyManager {
+  private _subscribed: MdBasicChip[] = [];
```

Comment:
I also think a `WeakMap` would be better for what you're doing (`WeakSet` would be best, but that's not supported in IE11).